### PR TITLE
Switch all instances of /obj/item/plank to /obj/item/sheet/wood, + remove /obj/item/plank

### LINKED ISSUE #15845: New wood material can't be used to change reinforced tables
### Labels

[BUG][MINOR]

### Description

If you have a wooden plank from the cargo warehouse, you can click on a reinforced table to make it into a fancy wooden table (like you see in the bar). If you use wood cut from a tree, it does not work.

### Steps to reproduce

1. Grow and cut down a tree for wood.
2. Attempt to use the wood on a reinforced table.

### Expected Behavior

Previously, wood cut from a tree was able to be used on reinforced tables to change the appearance.

### Additional Information & Screenshots

_No response_

### PR DESCRIPTION
**Pull request by MeggalBozale**
[QoL] [Game-Objects]
## About the PR 
Title, the functionality of the planks is now in wood sheets, and all instances of planks in game are changed. The specific changes are listed below. (I'm pretty sure it would goof something up if I _deleted_ the type since they're probably still around in secret content / maps, but I'd like to follow up on that in a future PR.)

Updates the random spawner to spawn wood sheets instead of planks.
Updates zombie code to use /obj/item/sheet/wood/zwood for constructing barricades. Probably the larger share of changes is for this behavior.
Allows wood sheets to be used to furnish reinforced tables in place of planks. Gives botany trees a nice use that should exist.
Replaces planks from digging on unsimmed dirt with wood sheets: It works, I tried it, shouldn't affect gameplay much.
Player pianos no longer varedit planks and name them 'Piano Maintenence Panel" - Instead any single sheet of plank will do.
Removes references to planks _Everywhere_ and deletes plank.dm. It be gone now
## Why's this needed? 
Primarily intended to fix #15845, but I also believe that bringing the two intuitively 'same-y' types together should help overall for gameplay. Lets tree wood be used for more things, all that.

## Changelog
```changelog
(u)MeggalBozale
(+)Wooden planks are mostly replaced with instances of wood sheets, meaning spare planks can be used to furnish tables now.
```

### CODE REVIEW

**Comment by MeggalBozale** [comment]
Not sure if I should find+replace all of the stuff in maps in this PR though id like to

**Comment by Glamurio** [comment]
> Not sure if I should find+replace all of the stuff in maps in this PR though id like to

I think if you're doing a "switch all instances" PR, you should switch ALL instances, and not only most of them. So yeah, go ahead. Don't worry about secret stuff, feel free to delete the obsolete stuff, some dev can handle those replacements.

**Comment by MeggalBozale** [comment]
Gocha. Worst case I'll revert the commit find+replaces all.

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Thanks for doing this, good cleanup 👍 
Would be good to retain the left click on turf to build barrier behaviour from planks.
Will need (very minor) secret changes.

File: code/datums/manufacturing.dm [hunk 1]
```diff
@@ -3515,11 +3515,11 @@ ABSTRACT_TYPE(/datum/manufacture/pod/weapon)
 	create = 1
 	category = "Ammo"
 /datum/manufacture/plank	//
-	name = "Plank/Barricade"
-	item_paths = list("MET-1")
+	name = "Barricade Planks"
+	item_paths = list("MET-2")
```
[the remark this answers is not in the document]
Comment (MeggalBozale) [reply code/datums/manufacturing.dm:3537]: Seems "WOOD" is wood, I'll change it to that

File: code/datums/manufacturing.dm [hunk 2]
```diff
@@ -3515,11 +3515,11 @@ ABSTRACT_TYPE(/datum/manufacture/pod/weapon)
...
 	item_amounts = list(1)
-	item_outputs = list(/obj/item/plank/anti_zombie)
-	time = 5 SECONDS
+	item_outputs = list(/obj/item/sheet/wood/zwood)
+	time = 1 SECOND
```
[the remark this answers is not in the document]
Comment (MeggalBozale) [reply code/datums/manufacturing.dm:3540]: You can make barricades with the zwood, point is to remove _all_ of the planks even if they're a part of that zombie gamemode

File: code/obj/table.dm [hunk 3]
```diff
@@ -229,25 +231,17 @@ TYPEINFO_NEW(/obj/table)
...
-						qdel(src)
-						boutput(user, "<span class='notice'>You have added a faux wood finish to \the [src]</span>")
-					return
-				else
-					boutput(user, "<span class='notice'>\The [src] is too weak to be modified!</span>")
-			else
+		else if (istype(W,/obj/item/sheet/wood))
+			if (istype(src, /obj/table/reinforced/bar)) //why must you be so confusing
+				boutput(user, "<span class='notice'>You can't add more than one finish, that's just illogical!</span>")
+				return
```
[the remark this answers is not in the document]
Comment (MeggalBozale) [reply code/obj/table.dm:237]: Hm ok

**Pull request merged**